CMakeLists.txt

```diff
@@ -195,12 +195,26 @@ else()
   set(DCCL_HAS_B64 "0")
 endif()
 
-if(${PROTOC_VERSION} VERSION_LESS 2.4.0})
+# Protobuf < 2.4.0 does not have plugin capability
+if(${PROTOC_VERSION} VERSION_LESS 2.4.0)
   option(enable_units "Enable static unit-safety functionality" OFF)
 else()
   option(enable_units "Enable static unit-safety functionality" ON)
 endif()
 
+# Protobuf >= 3.0.0. requires a syntax version
+if(NOT ${PROTOC_VERSION} VERSION_LESS 3.0.0) 
+  set(PROTOBUF_SYNTAX_VERSION "syntax = \"proto2\";")
+endif()
+
+# Protobuf >= 3.6.0 requires C++11 or greater
+if((NOT ${PROTOC_VERSION} VERSION_LESS 3.6.0)
+    AND ((NOT DEFINED CMAKE_CXX_STANDARD) OR ("${CMAKE_CXX_STANDARD}" STREQUAL "98")))
+  message ("Setting CMAKE_CXX_STANDARD to 11 as required by Protobuf >= 3.6.0. If you wish to use a newer standard, set CMAKE_CXX_STANDARD directly")
+  set(CMAKE_CXX_STANDARD 11)
```

Comment:
Note, `set(CMAKE_CXX_STANDARD 11)` requires CMake >= 3.1 but `CMakeLists.txt` specifies `cmake_minimum_required(VERSION 2.8.0 FATAL_ERROR)`

https://stackoverflow.com/questions/10851247/how-to-activate-c-11-in-cmake#comment49676678_10851247

Reply:
I think the odds of someone using CMake < 3.1 but Protobuf >= 3.6.0 are low enough to make this acceptable.

Reply:
I'm ok with it.